lib/expand.js

```diff
@@ -695,6 +697,47 @@ class Expander {
     return constraints;
   }
 
+  consolidateSubsetConstraint(element, value, constraint, previousConstraints) {
+    constraint = constraint.clone();
+    // If the constraint is actually on the target's value, update the path to explicitly include the target's value
```

Comment:
Note that the code below doesn't actually do what the comment says (update the path).  Since we're dealing with choices, I don't think it *can* update the path unless it put a keyword `Value` in there (but I don't remember if we support that anyway).